subsys/net/openthread/rpc/client/ot_rpc_commissioning_client.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <ot_rpc_ids.h>
+#include <ot_rpc_types.h>
+#include <ot_rpc_common.h>
+
+#include <nrf_rpc/nrf_rpc_serialize.h>
+#include <nrf_rpc/nrf_rpc_cbkproxy.h>
+
+static void ot_rpc_thread_discover_cb_rpc_handler(const struct nrf_rpc_group *group,
+						  struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	otHandleActiveScanResult callback_slot;
+	otActiveScanResult aResult;
+	struct ot_rpc_active_scan_result result;
+	void *pkt_data = &result;
+	void *context;
+
+	pkt_data = nrf_rpc_decode_buffer(ctx, pkt_data, sizeof(result));
```

Comment:
This doesn't check if the payload length is exactly `sizeof(result)`, only if it's no bigger than that.

Reply:
it was redesigned to separate cbor types. not actual anymore